cmd/release/main.go

```diff
@@ -26,123 +23,170 @@ import (
 	"aead.dev/minisign"
 	"github.com/bufbuild/buf/private/bufpkg/bufplugin/bufpluginref"
 	"github.com/google/go-github/v48/github"
-	"github.com/hashicorp/go-retryablehttp"
 	"golang.org/x/mod/semver"
-	"golang.org/x/oauth2"
 
 	"github.com/bufbuild/plugins/internal/plugin"
+	"github.com/bufbuild/plugins/internal/release"
 )
 
-const (
-	githubOwner = "bufbuild"
-	// This is separate from githubOwner for testing releases (can point at personal fork).
-	githubReleaseOwner = githubOwner // "pkwarren"
-	githubRepo         = "plugins"
-	pluginReleasesFile = "plugin-releases.json"
-)
-
-type ReleaseStatus int
-
-const (
-	EXISTING ReleaseStatus = iota
-	NEW
-	UPDATED
-)
-
-type PluginReleases struct {
-	Releases []PluginRelease `json:"releases"`
-}
-
-type PluginRelease struct {
-	PluginName       string        `json:"name"`           // org/name for the plugin (without remote)
-	PluginVersion    string        `json:"version"`        // version of the plugin (including 'v' prefix)
-	PluginZipDigest  string        `json:"zip_digest"`     // <digest-type>:<digest> for plugin .zip download
-	PluginYAMLDigest string        `json:"yaml_digest"`    // <digest-type>:<digest> for buf.plugin.yaml
-	ImageID          string        `json:"image_id"`       // <digest-type>:<digest> - https://github.com/opencontainers/image-spec/blob/main/config.md#imageid
-	RegistryImage    string        `json:"registry_image"` // ghcr.io/bufbuild/plugins-<org>-<name>@<digest-type>:<digest>
-	ReleaseTag       string        `json:"release_tag"`    // GitHub release tag - i.e. 20221121.1
-	URL              string        `json:"url"`            // URL to GitHub release zip file for the plugin - i.e. https://github.com/bufbuild/plugins/releases/download/20221121.1/bufbuild-connect-go-v1.1.0.zip
-	LastUpdated      time.Time     `json:"last_updated"`
-	Status           ReleaseStatus `json:"-"`
-}
-
 type pluginNameVersion struct {
 	name, version string
 }
 
 func main() {
-	var (
-		minisignPrivateKey string
-		dryRun             bool
+	dryRun := flag.Bool("dry-run", false, "perform a dry-run (no GitHub modifications)")
+	githubReleaseOwner := flag.String(
+		"github-release-owner",
+		string(release.GithubOwnerBufbuild),
+		"GitHub release owner (set to personal account to test against a fork)",
+	)
+	minisignPrivateKey := flag.String("minisign-private-key", "", "path to minisign private key file")
+	minisignPublicKey := flag.String(
+		"minisign-public-key",
+		"",
+		"path to public key used to verify the latest release's plugin-releases.json file (if different than private key)",
```

Comment:
This arg is used if the public key used to verify the plugin-releases.json file doesn't match the private key used to sign (this might be if you're using `-dry-run` locally).